### PR TITLE
Add minor and patch release information.

### PR DESCRIPTION
**Pull request by dblock**
### Description

Did a pass on RELEASING. We have some information in https://opensearch.org/releases.html such as the fact that we follow OpenSSF patch criteria, and don't explicitly mention minor and patch releases containing features/fixes/security patches. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: RELEASING.md
Comment (seraphjiang) [inline RELEASING.md:3]: how about other repo

File: RELEASING.md
Comment (seraphjiang) [inline RELEASING.md:10]: in our 2024 roadmap, we plan to decouple opensearch release and opensearch dashboards release. would suggest to check that and think through if we should define process to couple these two

File: RELEASING.md
Comment (seraphjiang) [inline RELEASING.md:44]: where is the branch for next patch release? could we add that, i would suggest to use 2.0.x

File: RELEASING.md
Comment (seraphjiang) [inline RELEASING.md:51]: again, OSD release should decouple from opensearch release in the furture(2025), plugin release should decoupled with OpenSearch and OSD as well. at least OSD

File: RELEASING.md
Comment (seraphjiang) [inline RELEASING.md:63]: missing critical bug for patch version release. could we add?
also do we mentioned critical for security and critical bug

File: RELEASING.md
Comment (seraphjiang) [inline RELEASING.md:77]: let's clarify what is build number. typically it is to identify build, ideally we should have
2.9.0.(12345) as full version number.

12345 could be a build number generated by build automation and it is always increase.

File: RELEASING.md
Comment (seraphjiang) [inline RELEASING.md:92]: let's call out the limitation of this appraoch, right now the distribution release bundled OpenSearch, OpenSearch Dashboards and plugin. In the case OpenSearch doesn't has major change, OSD has break change, it is not clear if we should let OpenSearch Dashboards hold the critical feature(with break) to release and wait for OpenSearch. Especially the break change is caused by Security issue like CVE.

File: RELEASING.md
Comment (seraphjiang) [inline RELEASING.md:102]: For OSD 1.3.x is still Node 10 , its lifecycle Ended 3 years and 4 months ago. we need a better version strategy and lifecycle management for long term support. in the case. i would suggest to decouple OpenSearch Dashboards and OpenSearch release. and we could end support to OSD 1.3 because it is not secure, and we could not upgrade without break. 

https://github.com/opensearch-project/OpenSearch-Dashboards/blob/d0b47bdab4e5d5df4ea1e80b6fb928cb027824cf/package.json#L511

File: RELEASING.md
Comment (seraphjiang) [inline RELEASING.md:112]: we did 6 minor version release in 2023, planed 6 minor in 2024, so it is about every 8 weeks right.

File: RELEASING.md
Comment (seraphjiang) [inline RELEASING.md:123]: what's the different between final quality test and test we enforce for each commit/PR

File: RELEASING.md
Comment (seraphjiang) [inline RELEASING.md:152]: could we elaborate why we choose to tag without `v`, it is very confusion compare to branch. large oss project like linux, react use v- for tag

https://github.com/torvalds/linux/tags
https://github.com/facebook/react/tags

File: RELEASING.md
Comment (seraphjiang) [inline RELEASING.md:144]: i'm glad to see we remove this

**Comment by seraphjiang** [comment]
hi repo/org admin
this this reviewer list still update to date?

image

File: RELEASING.md
Comment (getsaurabh02) [reply RELEASING.md:10]: This is as part current architecture. @seraphjiang Once we have proposal for decoupling accepted and executed for the project - we can revisit then.

File: RELEASING.md
Comment (getsaurabh02) [reply RELEASING.md:44]: How will that be helpful?

File: RELEASING.md
Comment (getsaurabh02) [reply RELEASING.md:63]: OpenSearch follows [Semver](https://semver.org/) - ```PATCH version when you make backward compatible bug fixes```

File: RELEASING.md
Comment (getsaurabh02) [reply RELEASING.md:51]: Can you share the proposal here?

File: RELEASING.md
Comment (getsaurabh02) [inline RELEASING.md:100]: Critical bugs should be limited to existing features (not newly introduced in the current minor release), where users who are already utilizing the feature must rollback due to issues such as broken functionality, significant performance degradation, or unexpected failures rendering the feature unusable.

Comment (getsaurabh02) [reply RELEASING.md:100]: It's important to emphasize existing features (not newly introduced) so that feature/component owners understand that incomplete or poorly developed features should not rely on the possibility of a patch release for completion. Also, with new feature we are not breaking user expectation, so it doesn't demand patch release either.

File: RELEASING.md
Comment (dblock) [reply RELEASING.md:3]: right below.

File: RELEASING.md
Comment (dblock) [reply RELEASING.md:44]: I added

```
* * **2.17**: The _last minor_ release and the _next patch_ release. Once a change is merged into `2.x`, it may be backported to, for example, `2.17` for the next 2.17 release candidate, or as a patch for a `2.12.1`.
```

File: RELEASING.md
Comment (dblock) [reply RELEASING.md:51]: we can update this when this happens

File: RELEASING.md
Comment (dblock) [reply RELEASING.md:63]: @seraphjiang I don't think I follow, can you please use the suggest button and add the text you think we should have here?

File: RELEASING.md
Comment (dblock) [reply RELEASING.md:77]: These are always 0 in OpenSearch, so I don't think this is correct.

File: RELEASING.md
Comment (dblock) [reply RELEASING.md:92]: I changed it to an example.

_For example, the OpenSearch bundle that contains OpenSearch and OpenSearch Dashboards releases new major versions after Lucene major version releases. This is announced in the forums at least 4 weeks prior to the release date. Once we become aware of the need for a major version, we start a major version release window which is similar to a minor release, except for two things: 1) participation is mandatory for all components, and 2) the release window is at least 4 weeks long to accommodate the testing required._

File: RELEASING.md
Comment (dblock) [reply RELEASING.md:102]: Anything that this PR needs changing?

File: RELEASING.md
Comment (dblock) [reply RELEASING.md:112]: @getsaurabh02 is this correct?

File: RELEASING.md
Comment (dblock) [reply RELEASING.md:123]: Not sure. Suggest text?

File: RELEASING.md
Comment (dblock) [reply RELEASING.md:152]: Below it says

_For a discussion on whether to add a prefixing `v` to release tags, see [#35](https://github.com/opensearch-project/.github/issues/35)._

**Comment by dblock** [comment]
> hi repo/org admin this this reviewer list still update to date?
> 
> image

Yes it is, why?

File: RELEASING.md
Comment (dblock) [reply RELEASING.md:100]: Can you please suggest some text that I can copy-paste?

File: RELEASING.md
Comment (seraphjiang) [reply RELEASING.md:64]:
```suggestion
Backwards-incompatible changes always result in a new major version and will __never__ be backported. Small improvements and features will be backported to a new minor version (e.g. `2.9.0`). Security or critical bug fixes will be backported to a new patch version (e.g. `2.9.1`). Repos in this organization typically use a backport workflow where you can label a PR, e.g. `backport 2.x`, and the workflow will attempt an automatic backport and open a new PR. If the backport fails, it is the contributor's responsibility to do a manual backport following the instructions in the failed backport error message.
```

File: RELEASING.md
Comment (seraphjiang) [reply RELEASING.md:44]: @dblock 

* * **2.x**: The _next minor_ release
`2.17` for the next 2.17 release candidate, or as a patch for a `2.12.1`.


what's the different between _next minor_ releae compare to release candidate?

if 2.x if for next minor release, why could we add 2.17.x for next patch release, then we should use 2.17.1 for release candidate 

i know opensearch is using 2.17.1 for tag. but it is not very common compare to other opensource project.

File: RELEASING.md
Comment (seraphjiang) [reply RELEASING.md:78]: if build name is not correct, should we use below 
```suggestion
The version number of all releasable artifacts is 3-digit `major.minor.patch` (e.g. `2.9.0`).
```

File: RELEASING.md
Comment (seraphjiang) [reply RELEASING.md:101]: how about this

```suggestion
Patch releases are reserved for addressing high-severity CVEs, critical bug fixes (e.g., issues causing users to roll back after an upgrade), or significant regressions in non-experimental features (e.g., feature rendered unusable). These releases must not introduce any new features.

Whether a fix qualifies as a critical bug should be proposed by the PR contributor or issue creator and triaged by the maintainer, with input from the community. The severity and impact of a bug in defining a critical issue can vary based on the type of repository. For example, a UX bug might be critical for end-user customers and could prevent them from using a feature, requiring domain expertise to assess. In contrast, backend repository bugs are generally more straightforward to triage.
```

File: RELEASING.md
Comment (seraphjiang) [reply RELEASING.md:103]:
```suggestion
We follow [OpenSSF's best practices](https://bestpractices.coreinfrastructure.org/en/criteria/0?details=true&rationale=true#0.vulnerabilities_fixed_60_days) for patching publicly known vulnerabilities and we make sure that there are no unpatched vulnerabilities of medium or higher severity that have been publicly known for more than 60 days in our actively maintained versions.

Please note: The runtime for OpenSearch Dashboards 1.3.x is based on Node.js 10, which has been end-of-life (EOL) for 3 years. Due to this, critical bug fixes and security patches are handled on a best-effort basis, as upgrading Node.js for 1.3.x would introduce significant breaking changes. This limitation arises from the need to release OpenSearch and OpenSearch Dashboards together.

We are aware of this issue and have a roadmap item to decouple the releases and versions for OpenSearch and OpenSearch Dashboards. This will allow us to upgrade or manage the versions of OpenSearch Dashboards independently in the future. More details can be found in the [OpenSearch Project Roadmap for 2024–2025](https://opensearch.org/blog/opensearch-project-roadmap-2024-2025/#roadmap-theme-3-ease-of-use) section: Serverless Dashboards and migration. 
```

File: RELEASING.md
Comment (seraphjiang) [reply RELEASING.md:124]:
```suggestion
2. During the release window, we will conduct final quality testing, documentation updates, and performance testing. Bug fixes can be added during this time, but no new features will be included. We are not yet able to run release testing against every commit of each repository. During the release process, we will only enforce that each component addresses failed tests by either fixing them or skipping flaky tests. Neither the 2.x nor the 2.17 branches will meet the release quality bar without going through the full release process, including testing. In the event that any third party wants to fork and release from the 2.x or 2.17 branches, they will need to ensure they run their own quality tests at their own risk.
```

File: RELEASING.md
Comment (seraphjiang) [reply RELEASING.md:153]: Thanks, @dblock. It doesn't explain why we chose not to add the "v-" at the beginning, but I understand it was a concern. there are effort to support, but no one work on it. and the world have gotten used to it now. 

Good to know, but we should try in new repository. Especially for those repository we took the ownership for maintenance from other open source org. like fork new repo.

File: RELEASING.md
Comment (dblock) [reply RELEASING.md:103]: This is way too specific for this doc. What's the GitHub issue that captures this detail? 

I could change "we follow" to "Except for well-documented exceptions such as xyz, we follow ...".

File: RELEASING.md
Comment (dblock) [reply RELEASING.md:124]: I think this is confusing, at least in the release window section because it talks about state of a 2.x branch. I'll think about where to incorporate it, but for now not adding.

File: RELEASING.md
Comment (dblock) [reply RELEASING.md:101]: Taking first paragraph.

For the second I shortened a bit

_Whether a fix qualifies as a critical bug should be proposed by the PR contributor or issue creator, and triaged by the maintainers of the repo with input from the community. The severity and impact of a bug in defining criticality of an issue can vary. For example, back-end APIs are fairly straightforward to triage, while a UX bug may prevent users from using a feature and may require some domain expertise to assess._

File: RELEASING.md
Comment (dblock) [reply RELEASING.md:44]: On 2.x you usually have the next minor release. Then eventually we cut 2.17 and make last minute fixes there to make the 2.17 release. So they are the same thing.

Then we make patches on top of 2.17 in the 2.17 branch while 2.x will have moved to 2.18. And release 2.17.1 from 2.17 branch.

Suggest a change if that's not clear from my text?

File: RELEASING.md
Comment (seraphjiang) [reply RELEASING.md:103]: I linked roadmap, a github issue might not necessary, if words is too much, we keep it short? there are two keypoints here, first is security patch is best effort for previous major version (1.x or what term we name it, please help). second it doesn't scale if we bump whole bundle distribution if only one component need to bump. decoupling project and make them release-able independently is roadmap.

**Comment by dblock** [comment]
I think this can be merged as is, wdyt @peternied @CEHENKLE?

File: RELEASING.md
Comment (CEHENKLE) [inline RELEASING.md:68]: Any regression that breaks semver?  Also, [according to the spec](https://semver.org/) it's "SemVer"

File: RELEASING.md
Comment (CEHENKLE) [inline RELEASING.md:93]: So whaddya do if a component doesn't participate?  Drop it?   With some components down to a few maintainers, this is a going concern.

**Review by CEHENKLE — APPROVED** [review]
Left a few nits, but generally moves things forward :)

File: RELEASING.md
Comment (dblock) [reply RELEASING.md]: Fixed both.

File: RELEASING.md
Comment (dblock) [reply RELEASING.md:93]: The release team picks it up.

File: RELEASING.md
Comment (peternied) [inline RELEASING.md:68]: I like taking a firm stance on this, also this isn't being followed in all repositories - opensearch-migration that I maintain isn't following semver.  

Do we have thoughts on how we roll out this **P**olicy change is known?

**Review by peternied — APPROVED** [review]
Thank for making this update, looks good to me.

File: RELEASING.md
Comment (dblock) [reply RELEASING.md:68]: AFAIK this has been the policy. I think this is a bug in opensearch-migrations, you should open it!

Comment (peternied) [reply RELEASING.md:68]: - https://github.com/opensearch-project/opensearch-migrations/issues/1042

**Comment by dblock** [comment]
@peternied I rebased.

**Comment by peternied** [comment]
@seraphjiang There are a couple of outstanding threads from feedback you created on this change, did you want to take another look while these changes bake?

**Comment by peternied** [comment]
@CEHENKLE @getsaurabh02 We are waiting on a second approval, what do you think about merging this change?

**Review by CEHENKLE — APPROVED** [review]

**Pull request merged**
